src/groupeddataframe/splitapplycombine.jl

```diff
@@ -855,11 +855,45 @@ end
 
 # Use reducedim_init to get a vector of the right type,
 # but trick it into using the expected length
-groupreduce_init(op, condf, incol, gd) =
-    Base.reducedim_init(identity, op, view(incol, 1:length(gd)), 2)
+function groupreduce_init(op, condf, adjust, incol, gd)
+    U = eltype(incol)
+    T = Base.promote_union(U)
+
+    if op isa typeof(Base.add_sum)
+        f = zero
+    elseif op isa typeof(Base.mul_prod)
+        f = one
+    else
+        throw(ErrorException("Unrecognized op $op"))
+    end
+
+    if T !== Any && applicable(f, T)
+        initv = f(T)
+        x = adjust isa typeof(/) ? (initv / 1) : initv
+        V = U >: Missing ? Union{typeof(x), Missing} : typeof(x)
+    else
+        idx = findfirst(!ismissing, incol)
+        if isnothing(idx)
+            x = missing
+        else
+            initv = f(incol[idx])
+            x = adjust isa typeof(/) ? (initv / 1) : initv
+        end
+        # do not try to determine the narrowest possible type as this is not
+        # possible to do correctly in general without processing groups
+        # it will get fixed later in groupreduce!
+        V = Any
+    end
+    v = Vector{V}(undef, length(gd))
```

Comment:
@nalimilan - just to be sure - it is impossible here that `v` should be a `CategoricalVector` here - right?
(I ask this because in code below I have seen that we check against `CategoricalVector` in reductions and I was not sure why).

Reply:
I have changed to `similar(incol, V, length(gd))` but it really I cannot see in practice the case where it will differ (a quick look at `methods` of `similar` shows that normally it will always produce `Vector`).